indico/web/forms/fields.py

```diff
@@ -383,6 +384,34 @@ def _get_data(self):
         return [] if self.data is None else [self.data]
 
 
+class PersonLinkListFieldBase(EventPersonListField):
+
+    #: class that inherits from `PersonLinkBase`
+    person_link_cls = None
+
+    widget = None
+
+    @no_autoflush
+    def _get_person_link(self, data, extra_data=None):
+        extra_data = extra_data or {}
+        person = self._get_event_person(data)
+        person_data = {'title': next((x.value for x in UserTitle if data.get('title') == x.title), UserTitle.none),
+                       'first_name': data.get('firstName', ''), 'last_name': data['familyName'],
+                       'affiliation': data.get('affiliation', ''), 'address': data.get('address', ''),
+                       'phone': data.get('phone', '')}
+        person_data.update(extra_data)
+        return self.person_link_cls(person=person, **person_data)
+
+    def _serialize_principal(self, principal):
+        if not isinstance(principal, PersonLinkBase):
+            return super(PersonLinkListFieldBase, self)._serialize_principal(principal)
+        else:
+            return self._serialize_person_link(principal)
+
+    def _serialize_person_link(self, principal, extra_data=None):
+        raise NotImplemented
```

Comment:
that's not an exception (but something to return from certain magic comparison functions) - you want `NotImplementedError` instead